src/coredump/coredump.c

```diff
@@ -1736,11 +1781,9 @@ static int process_kernel(int argc, char* argv[]) {
                 if (r >= 0)
                         return 0;
 
-                r = gather_pid_mount_tree_fd(&context);
-                if (r < 0 && r != -EBADF)
+                r = gather_pid_mount_tree_fd(&context, &mount_tree_fd);
+                if (r < 0)
                         log_warning_errno(r, "Failed to access the mount tree of a container, ignoring: %m");
```

Comment:
you already log about all errors inside of gather_pid_mount_tree_fd(), hence this is strictly speaking duplicate logging.

i think it makes sense to keep a log message in place here however, if it tells us *new* stuff the original log messages inside of gather_pid_mount_tree_fd() didn't tell us yet. Specifically, it's the "ignoring" part which the function didn't mention. So maybe keep the log message mostly, but drop the %m (and change from log_warning_errno() to log_warning()).

that way, we will log about the error code once, and then log about the action we are taking because of it. (which is "no action")